yt_dlp/extractor/dailymotion.py

```diff
@@ -383,9 +387,61 @@ def _extract_embed_urls(cls, url, webpage):
+
+        if not self._HEADERS.get('Authorization'):
+            self._HEADERS['Authorization'] = f'Bearer {self._get_token(term)}'
+        resp = self._download_json(
+            'https://graphql.api.dailymotion.com/', None, note, data=json.dumps({
+                'operationName': 'SEARCH_QUERY',
+                'query': search_query,
+                'variables': {
+                    'limit': 20,
+                    'page': page,
+                    'query': term,
+                    'shouldIncludeChannels': False,
+                    'shouldIncludeLives': False,
+                    'shouldIncludePlaylists': False,
+                    'shouldIncludeTopics': False,
+                    'shouldIncludeVideos': True
+                }
+            }).encode(), headers=self._HEADERS)
+        obj = traverse_obj(resp, ('data', 'search', {dict}))
+        if not obj:
+            raise ExtractorError(traverse_obj(resp, ('errors', 0, 'message'))
+                                 or 'Could not fetch search data', expected=True)
```

Comment:
```suggestion
            raise ExtractorError(
                traverse_obj(resp, ('errors', 0, 'message', {str})) or 'Could not fetch search data', expected=True)
```
Why is this expected, though? (I'm just asking a question, not necessarily suggesting anything other than the formatting be changed)

Reply:
Yes, I think this shouldn't be expected. Empty search results do not raise this error.